docs/hibernating-clusters.md

```diff
@@ -0,0 +1,121 @@
+# Hibernating Clusters
+
+## Overview
+
+Starting with OpenShift version 4.4.8, clusters can be stopped and started by simply shutting
+down their machines and starting them back up. The only additional requirement to bring a cluster
+back up from a stopped state is that CSRs be approved for the cluster's nodes when certificates
+have expired since the cluster was active.
+
+Hive can automate the process of stopping/starting clusters via its API by allowing the user to
+set a the desired state of the cluster in the ClusterDeployment spec. Both API and controller
+changes are required to support this feature.
+
+## API Changes
+
+The ClusterDeploymentSpec should allow setting whether machines are in a running state or in
+a hibernating state.
+
+```go
+type ClusterDeploymentSpec struct {
+  // ... other fields
+
+  // Hibernating indicates whether a cluster should be hibernating. Default is false.
+  Hibernating  `json:"hibernating,omitempty"`
+}
+```
+
+On the status side, a cluster's state is reflected by the following condition types:
+
+```go
+const (
+  // ... other constants
+
+  // ClusterStartingConditionType is set when the ClusterDeployment is transitioning
+  // from a hibernating state to a running state.
+  ClusterStartingConditionType ClusterDeploymentConditionType = "ClusterStarting"
+
+  // ClusterRunningConditionType is set when the ClusterDeployment is in a running and
+  // ready state.
+  ClusterRunningConditionType ClusterDeploymentConditionType = "ClusterRunning"
+
+  // ClusterStoppingConditionType is set when the ClusterDeployment is transitioning
+  // from a running state to a hibernating state.
+  ClusterStoppingConditionType ClusterDeploymentConditionType = "ClusterStopping"
```

Comment:
This seems like a lot of conditions to add to the ClusterDeployment. Can we accomplish the same thing with reasons on a single "ClusterHibernating" condition.

Original | Status | Reason       
--- | --- | ---
ClusterStarting | True | Resuming
ClusterRunning | False | Running
ClusterStopping | True | Stopping
ClusterHibernating | True | Hibernating
HibernationNotSupported | False | NotSupported

Reply:
I like it. I was torn with the multiple conditions, but still liked the benefit of a condition such as knowing when it last transitioned. This looks better.

Reply:
ClusterRunning could be used for a "HibernateAfter" feature so we could always power down after 8h which might be useful for a lot of use cases. If it's just one condition though it's still possible as that would be the current condition.